distributed/scheduler.py

```diff
@@ -2220,9 +2220,6 @@ def decide_worker_non_rootish(self, ts: TaskState) -> WorkerState | None:
 
         valid_workers = self.valid_workers(ts)
         if valid_workers is None and len(self.running) < len(self.workers):
-            if not self.running:
-                return None
```

Comment:
Is this related? At least the new test doesn't seem to care about this.

Reply:
It's unreachable because the same condition is already tested on line 2218